feature_engine/encoding/count_frequency.py

```diff
@@ -1,6 +1,7 @@
 # Authors: Soledad Galli <solegalli@protonmail.com>
 # License: BSD 3 clause
 
+from collections import defaultdict
```

Comment:
It's a different type of dictionary that outputs zeros for entries that it doesn't know. I think it is a very good candidate to use for this PR since it makes it involve very little changes, and it's probably faster than doing checks after encoding.